internal/service/cluster/model_cluster.go

```diff
@@ -0,0 +1,528 @@
+		if err != nil {
+			return regionsConfig, err
+		}
+
+		if replaceRegion != "" && r == originalRegion {
+			r, err = conversion.ValRegion(replaceRegion)
+		}
+		if err != nil {
+			return regionsConfig, err
+		}
+
+		regionsConfig[r] = matlas.RegionsConfig{
+			AnalyticsNodes: conversion.Pointer(cast.ToInt64(region["analytics_nodes"])),
+			ElectableNodes: conversion.Pointer(cast.ToInt64(region["electable_nodes"])),
+			Priority:       conversion.Pointer(cast.ToInt64(region["priority"])),
+			ReadOnlyNodes:  conversion.Pointer(cast.ToInt64(region["read_only_nodes"])),
+		}
+	}
+
+	return regionsConfig, nil
+}
+
+func flattenRegionsConfig(regionsConfig map[string]matlas.RegionsConfig) []map[string]any {
+	regions := make([]map[string]any, 0)
+
+	for regionName, regionConfig := range regionsConfig {
+		region := map[string]any{
+			"region_name":     regionName,
+			"priority":        regionConfig.Priority,
+			"analytics_nodes": regionConfig.AnalyticsNodes,
+			"electable_nodes": regionConfig.ElectableNodes,
+			"read_only_nodes": regionConfig.ReadOnlyNodes,
+		}
+		regions = append(regions, region)
+	}
+
+	return regions
+}
+
+func expandProviderSetting(d *schema.ResourceData) (*matlas.ProviderSettings, error) {
+	var (
+		region, _          = conversion.ValRegion(d.Get("provider_region_name"))
+		minInstanceSize    = getInstanceSizeToInt(d.Get("provider_auto_scaling_compute_min_instance_size").(string))
+		maxInstanceSize    = getInstanceSizeToInt(d.Get("provider_auto_scaling_compute_max_instance_size").(string))
+		instanceSize       = getInstanceSizeToInt(d.Get("provider_instance_size_name").(string))
+		compute            *matlas.Compute
+		autoScalingEnabled = d.Get("auto_scaling_compute_enabled").(bool)
+		providerName       = cast.ToString(d.Get("provider_name"))
+	)
+
+	if minInstanceSize != 0 && autoScalingEnabled {
+		if instanceSize < minInstanceSize {
+			return nil, fmt.Errorf("`provider_auto_scaling_compute_min_instance_size` must be lower than `provider_instance_size_name`")
+		}
+
+		compute = &matlas.Compute{
+			MinInstanceSize: d.Get("provider_auto_scaling_compute_min_instance_size").(string),
+		}
+	}
+
+	if maxInstanceSize != 0 && autoScalingEnabled {
+		if instanceSize > maxInstanceSize {
+			return nil, fmt.Errorf("`provider_auto_scaling_compute_max_instance_size` must be higher than `provider_instance_size_name`")
+		}
+
+		if compute == nil {
+			compute = &matlas.Compute{}
+		}
+		compute.MaxInstanceSize = d.Get("provider_auto_scaling_compute_max_instance_size").(string)
+	}
+
+	providerSettings := &matlas.ProviderSettings{
+		InstanceSizeName: cast.ToString(d.Get("provider_instance_size_name")),
+		ProviderName:     providerName,
+		RegionName:       region,
+		VolumeType:       cast.ToString(d.Get("provider_volume_type")),
+	}
+
+	if d.HasChange("provider_disk_type_name") {
+		_, newdiskTypeName := d.GetChange("provider_disk_type_name")
+		diskTypeName := cast.ToString(newdiskTypeName)
```

Comment:
more of a general comment: I see cast.ToString/Bool/etc being used a lot and also see .(bool)/.(string)/etc being used. I think the latter is the more idiomatic way of casting in Go and in my opinion more clear. WDYT?  No need to change in this PR but I think we should probably only use one of the ways

Reply:
you're totally right, it's something i also discussed with @gssbzn that we're using both styles all across the code and we might want to unify